### PR TITLE
refactor: migrate TagsPoplist, TextBox, TimeInput, Toaster stories to tsx

### PR DESCRIPTION
**Pull request by QuiiBz**
## Type

- Refactor

#### The following changes where made:

Migrate TagsPoplist, TextBox, TimeInput, Toaster stories to tsx

### CODE REVIEW

**Review by vincentaudebert — COMMENTED** [review]
Ping me when changes have been done to remove decorators etc as mentionned during stand up 👌

File: src/components/TextBox/index.tsx [hunk 1]
```diff
@@ -340,7 +341,8 @@ type TextBoxProps = {
 } & (
   | Omit<InputHTMLAttributes<HTMLInputElement>, 'onChange'>
   | Omit<TextareaHTMLAttributes<HTMLTextAreaElement>, 'onChange'>
-) & XStyledProps
+) &
+  XStyledProps
 
 const TextBox = forwardRef<
```
Comment (vincentaudebert) [inline src/components/TextBox/index.tsx:347]: Est-ce qu'il faut preciser que ca renvoie un JSX.Element quelque part pour TS peut etre?

File: src/components/TimeInput/index.tsx [hunk 2]
```diff
@@ -197,9 +197,10 @@ const defaultValue = {
   value: '00:00',
 }
 
-type TimeInputProps = Partial<RichSelectProps> & {
+export type TimeInputProps = Partial<RichSelectProps> & {
   schedule?: ScheduleType
 }
+
 const TimeInput: FunctionComponent<TimeInputProps> & {
```
Comment (vincentaudebert) [inline src/components/TimeInput/index.tsx:204]: Nuke FunctionComponent to use JSX.Element

File: src/components/TagsPoplist/index.tsx [hunk 3]
```diff
@@ -32,7 +32,8 @@ const StyledTooltipReference = styled(TooltipReference)`
   padding-left: 8px;
   padding-right: 8px;
 `
-type TagsPoplistProps = {
+
+export type TagsPoplistProps = {
```
Comment (vincentaudebert) [inline src/components/TagsPoplist/index.tsx:36]: Same in this file, migrate to JSX.Element

**Review by vincentaudebert — CHANGES_REQUESTED** [review]
Might have missed some but please double check migration to JSX.Element too.
And story:  { description: { as per DS-51

**Review by vincentaudebert — APPROVED** [review]

File: src/components/TagsPoplist/__stories__/index.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    description: {
+      story:
+        '`maxTagWidth` can be used for defining max width of each tags except the ones in tooltip.',
+    },
+  },
+}
+
+TagWidth.args = {
+  maxTagWidth: 30,
+  tags: [
```
Comment (DorianMaliszewski) [inline src/components/TagsPoplist/__stories__/index.stories.tsx:107]: Something more thin please 😅 ex : `Array(100).fill('large')`

File: src/components/TagsPoplist/__stories__/index.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+  docs: {
+    description: {
+      story:
+        '`threshold` prop defines the number of tags to display before hiding them into a tooltip.',
+    },
+  },
+}
+
+Threshold.args = {
+  tags: [
```
Comment (DorianMaliszewski) [inline src/components/TagsPoplist/__stories__/index.stories.tsx:35]: Here too

**Review by DorianMaliszewski — CHANGES_REQUESTED** [review]

File: src/components/TagsPoplist/__stories__/index.stories.tsx (hunk 4 above)
Comment (QuiiBz) [reply src/components/TagsPoplist/__stories__/index.stories.tsx:58]: Thanks, initially copied from the mdx story 😅

**Review by DorianMaliszewski — APPROVED** [review]

**Pull request merged**